core/rollup_fee.go

```diff
@@ -0,0 +1,22 @@
+package core
+
+import (
+	"math/big"
+)
+
+/// ROLLUP_BASE_TX_SIZE is the encoded rollup transaction's compressed size excluding
+/// the variable length data.
+/// Ref: https://github.com/ethereum-optimism/contracts/blob/409f190518b90301db20d0d4f53760021bc203a8/contracts/optimistic-ethereum/OVM/precompiles/OVM_SequencerEntrypoint.sol#L47
+var ROLLUP_BASE_TX_SIZE int = 96
+
+/// CalculateFee calculates the fee that must be paid to the Rollup sequencer, taking into
+/// account the cost of publishing data to L1.
+/// Returns: (ROLLUP_BASE_TX_SIZE + len(data)) * dataPrice + executionPrice * gasUsed
+func CalculateRollupFee(data []byte, gasUsed uint64, dataPrice, executionPrice *big.Int) *big.Int {
+	dataLen := int64(ROLLUP_BASE_TX_SIZE + len(data))
+	// get the data fee
+	dataFee := new(big.Int).Mul(dataPrice, big.NewInt(dataLen))
+	executionFee := new(big.Int).Mul(executionPrice, big.NewInt(int64(gasUsed)))
```

Comment:
I don't think that there will ever be an overflow with `int64(gasUsed)` but you could get away with not needing to cast it with `new(big.Int).SetUint64(gasUsed)`